bindings/python/YarpUtilities/src/VectorsCollection.cpp

```diff
@@ -65,17 +66,23 @@ void CreateVectorsCollectionClient(pybind11::module& module)
         .def("connect", &VectorsCollectionClient::connect)
         .def("disconnect", &VectorsCollectionClient::disconnect)
         .def("get_metadata",
-        [](VectorsCollectionClient& impl) -> BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata
-        {
-            BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata metadata;
-            impl.getMetadata(metadata);
-            return metadata;
-        })
+             [](VectorsCollectionClient& impl)
+                 -> BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata {
+                 BipedalLocomotion::YarpUtilities::VectorsCollectionMetadata metadata;
+                 impl.getMetadata(metadata);
```

Comment:
You can add a check if getMetadata returns true or false. If false you may throw an error